### PR TITLE
bugfix: Show episodes filtering

### LINKED ISSUE #1: Show episodes appear instantly after uploading
_What's wrong:_ Show episodes appear in the archive before they've been aired.
_How it should work:_ The show episodes should be filtered before they appear in the archive. Not-yet-aired episodes should not be available to listen.

### PR DESCRIPTION
**Pull request by molnar-a**

### CODE REVIEW

File: src/components/ShowDetail/showDetail.tsx [hunk 1]
```diff
@@ -38,20 +38,22 @@ export const ShowDetail = (
...
-          return -1;
-        }
-        return 0;
-      }),
-    [items]
-  );
+  const orderedItems = useMemo(() => {
+    const availableItems = items.filter(item => {
+      const playDate = new Date(item.play_date);
+      return item.archived && playDate < new Date();
```
Comment (baxgas) [inline src/components/ShowDetail/showDetail.tsx]: @molnar-a sorry, but this doesn't work, you are comparing different types of Date formats.
The copy/paste fix from ananasz would be 
```
const availableItems = items.filter(item => {
      const d = new Date();
      const year = d.getFullYear();
      const month = (d.getMonth() + 1).toLocaleString('en-US', {
        minimumIntegerDigits: 2
      });
      const day = d
        .getDate()
        .toLocaleString('en-US', { minimumIntegerDigits: 2 });
      const today = `${year}-${month}-${day}`;
      return item.archived && item.play_date < today;
    });
```
(but of course this `today` to arcsi-style formatting snippet could be modularized)

**Review by baxgas — CHANGES_REQUESTED** [review]
Doesn't work, please check the date comparison 

`new Date()` gives back the date with time, and you need only the date, in the same format as `play_date` from arcsi

image

File: src/components/ShowDetail/showDetail.tsx (hunk 1 above)
Comment (molnar-a) [reply src/components/ShowDetail/showDetail.tsx]: Thanks! I've resolved this now

**Review by baxgas — APPROVED** [review]

**Comment by baxgas** [comment]
@molnar-a Yesss! Seems it is working perfectly – tried on devarcsi which has some episodes made for this eg.
```
{
        "number": 789,
        "name": "future episode 789",
        "description": "big beats so good",
        "play_date": "2025-04-05",
        "id": 766,
        "download_count": 0,
        "image_url": "789/learning_audio_classification_with_torso-future_episode_789.jpg",
        "archived": false,
        "play_file_name": null
      },
```
in Learning Audio...
Was filtered out nicely. Merge the branch please and it can go in app stores

**Pull request merged**